### PR TITLE
Integrate all access methods for this API into this repository

### PR DESCRIPTION
**Pull request by l0k18**
Currently the bls12_381-sign-go repository clones this one to build the binary objects needed for the Cgo and gRPC microservice. This will place the core FFI and Rust implementation of the signing API in one crate, the gRPC in another, and the Go package in a third subfolder, the Rust crates linked together in a workspace.

### CODE REVIEW

**Comment by l0k18** [comment]
- CI for Go tests needs to be reinstated

**Comment by l0k18** [comment]
when https://github.com/dusk-network/dusk-blockchain/pull/1183 is merged this should first be merged and that PR amended in the `go.mod` to then point to the new master branch where for now a makeshift commit tag is serving to specify this new location.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,44 @@
+all: installprotoc goprotos servicebinaries build test
+
+goprotos:
+	if [ $(shell uname -o) = "GNU/Linux" ]; \
+	then \
+		protoc --proto_path=./proto ./proto/bls12381sig.proto \
+			--go_opt=paths=source_relative \
+			--go_out=plugins=grpc:./bls/; \
+	else \
+		echo "not implemented yet for mac or windows"; exit; \
```
Comment (herr-seppia) [inline Makefile:10]: Makefile should support `macOS` too in the same way the old repo does https://github.com/dusk-network/bls12_381-sign-go/blob/main/Makefile

Comment (l0k18) [reply Makefile:10]: Isn't this in a separate issue? We were talking about @t00ts taking this task because he also has mac

**Comment by l0k18** [comment]
In addition to internal checks, https://github.com/dusk-network/dusk-blockchain/actions/runs/1452887500 shows that this branch at the current commit (`v0.1.0-merge3`) passes all tests on the primary (only?) consumer of this package, so it is now ok to merge this up to the main branch.

**Pull request merged**